lib/src/routing/connection_registry.rs

```diff
@@ -1,350 +1,221 @@
-            continue;
+    /// Mark a server as available for a specific database.
+    pub(crate) fn mark_unavailable(&self, server: &BoltServer) {
+        for db_name in self.databases.keys() {
+            debug!("Marking server as available: {server:?}");
+            let mut table = self.databases.get(&db_name).unwrap();
+            if table.mark_server_unavailable(server) {
+                self.databases.insert(db_name, table);
+                self.pool_registry
+                    .remove(&server.to_neo_url(self.scheme.as_str()));
+            } else {
+                debug!("Server not found in the registry: {server:?}");
+            }
         }
-        let uri = format!("{}://{}:{}", scheme, server.address, server.port);
-        debug!("Creating pool for server: {}", uri);
-        registry.insert(
-            server.clone(),
-            create_pool(&Config {
-                uri,
-                ..config.clone()
-            })?,
-        );
     }
-    debug!(
-        "Registry updated for database {}. New size is {} with TTL {}s",
-        db.as_ref().map_or("default".to_string(), |d| d.to_string()),
-        registry.len(),
-        routing_table.ttl
-    );
-    Ok(routing_table)
-}
-
-pub(crate) fn start_background_updater(
-    config: &Config,
-    registry: Arc<ConnectionRegistry>,
-    provider: Arc<dyn RoutingTableProvider>,
-) -> Sender<RegistryCommand> {
-    let config_clone = config.clone();
-    let (tx, mut rx) = mpsc::channel(1);
 
-    // This thread is in charge of refreshing the routing table periodically
-    tokio::spawn(async move {
-        let mut bookmarks = vec![];
-        let mut ttl = refresh_all_routing_tables(
-            config_clone.clone(),
-            registry.clone(),
-            provider.clone(),
-            bookmarks.as_slice(),
-        )
-        .await
-        .expect("Failed to get routing table. Exiting...");
-        debug!("Starting background updater with TTL: {}", ttl);
-        let mut interval = tokio::time::interval(Duration::from_secs(ttl));
-        let now = std::time::Instant::now();
-        interval.tick().await; // first tick is immediate
-        loop {
-            tokio::select! {
-                // Trigger periodic updates
-                _ = interval.tick() => {
-                    debug!("Refreshing all routing tables ({})", registry.databases.len());
-                    ttl = match refresh_all_routing_tables(config_clone.clone(), registry.clone(), provider.clone(), bookmarks.as_slice()).await {
-                        Ok(ttl) => ttl,
+    /// Get all available Bolt servers for a specific database or the default database if none is provided.
+    pub async fn servers(
+        &self,
+        db: Option<Database>,
+        imp_user: Option<ImpersonateUser>,
+        bookmarks: &[String],
+    ) -> Vec<BoltServer> {
+        if let Some(db_name) = db.as_ref().map(|d| d.to_string()) {
+            if let Some(table) = self.databases.get(&db_name) {
```

Comment:
With the change to `get` (see below), this can be changed to avoid the to_string:

```suggestion
        if let Some(db_name) = db.as_deref() {
            if let Some(table) = self.databases.get(db_name) {
```